compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
+#define FORWARD_LIST_HPP
+
+#pragma once
+
+#if 0
```

Comment:
I believe it's safe to extend STL containers as long as they destroyed using the most derived type.